### PR TITLE
Use the now-generalized reply executor for peer close.

### PR DESCRIPTION
**Pull request by JakeWharton**
This was feedback on https://github.com/square/okhttp/pull/1260 from @swankjesse

### CODE REVIEW

**Comment by JakeWharton** [comment]
One side-effect of this is that close might be delayed pending any queued pongs. Before this close would happen immediately and any queued pongs would just fail.

**Comment by swankjesse** [comment]
LGTM.

**Comment by JakeWharton** [comment]
This change might justify injecting the executor as it's what is causing the tests to fail.

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/ws/RealWebSocketTest.java [hunk 1]
```diff
@@ -50,7 +51,15 @@
   @Before public void setUp() {
     Random random = new Random(0);
     String url = "http://example.com/websocket";
-    client = new RealWebSocket(true, server2client, client2Server, random, clientListener, url) {
+
+    Executor synchronousExecutor = new Executor() {
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/ws/RealWebSocketTest.java]: Nice.

**Pull request merged**